browser/brave_rewards/rewards_tab_helper.cc

```diff
@@ -50,8 +50,14 @@ RewardsTabHelper::RewardsTabHelper(content::WebContents* web_contents)
   Profile* profile =
       Profile::FromBrowserContext(web_contents->GetBrowserContext());
   rewards_service_ = RewardsServiceFactory::GetForProfile(profile);
-  if (rewards_service_)
+  if (rewards_service_) {
     rewards_service_->AddObserver(this);
+#if BUILDFLAG(ENABLE_GREASELION)
+    rewards_service_->GetRewardsMainEnabled(
+        base::Bind(&RewardsTabHelper::GetRewardsMainEnabledCallback,
+                   base::Unretained(this)));
```

Comment:
I don't think this is safe for `base::Unretained` because the rewards service can outlive the tab helper

Reply:
but I don't think you need this anyway, I think this method probably needs to be deprecated cc @NejcZdovc because I'm pretty sure you can just check `kBraveRewardsEnabled`